Sources/SwiftParser/Directives.swift

```diff
@@ -192,7 +195,9 @@ extension Parser {
   ) -> [Element] {
     var elements = [Element]()
     var elementsProgress = LoopProgressCondition()
-    while !self.at(.eof) && !self.at(.poundElseKeyword, .poundElseifKeyword, .poundEndifKeyword) && !self.atElifTypo() && elementsProgress.evaluate(currentToken) {
+    while !self.at(.eof) && !self.at(.poundElseKeyword, .poundElseifKeyword, .poundEndifKeyword) && !self.atElifTypo()
+      && elementsProgress.evaluate(currentToken)
+    {
```

Comment:
I'm strong about this, but some places we place the `{` on a newline otherplaces we push each `&&` or `||` on a newline. 

IMO putting `&&` and `||` on a newline reads best.

Reply:
When the condition spans over multiple lines, swift-format will always put the `{` on a new line. Do you have an example where it doesn’t?

Reply:
Some places we add a newline for each condition like
```
   if node.conditions.count == 1,
      node.conditions.first?.as(ConditionElementSyntax.self)?.condition.is(MissingExprSyntax.self) == true,
      !node.body.leftBrace.isMissingAllTokens
    {
```
other places like the one above we add only a single && on a newline. 
IMO we should do like the code above so it should be maybe like

```
while !self.at(.eof)
      && !self.at(.poundElseKeyword, .poundElseifKeyword, .poundEndifKeyword)
      && !self.atElifTypo()
      && elementsProgress.evaluate(currentToken)
    {
```

Reply:
But I'm not strong about this, so if it doesn't make sense or you don't agree feel free to resolve.

Reply:
Just to be clear: You’re saying that you prefer `&&` over `,` wherever possible or are you trying to say something else?

Reply:
Nope 😅. I'm trying to say if one `&& someCondiion` or more is on a new line all off those should be on a new line.

So Thai while will be
```
while !self.at(.eof)
      && !self.at(.poundElseKeyword, .poundElseifKeyword, .poundEndifKeyword)
      && !self.atElifTypo()
      && elementsProgress.evaluate(currentToken)
    {

```

Same if there is a `,` or `|`.